tests/integration/test_smoke.py

```diff
@@ -285,15 +285,14 @@ def test_client_can_retry_request_properly():
 
 def _make_client_call_with_errors(client, operation_name, kwargs):
     operation = getattr(client, xform_name(operation_name))
-    original_send = adapters.HTTPAdapter.send
-    def mock_http_adapter_send(self, *args, **kwargs):
+    original_send = Endpoint._send
```

Comment:
Same comment as before, but we *really* shouldn't be touching internal methods in an integration test.

If we need to add better hooks into the endpoint to make it more testable via some public API, I'm fine with that.

Reply:
Yeah, I think this is another case for a before_send and after_send type of event.